internal/handler/command_test.go

```diff
@@ -1,36 +1,489 @@
+		{"NoDeviceResourceForResult", &deviceIntegerGenerator, "NoDeviceResourceForResult", paramsNoDeviceResourceForResult, true},
+		{"DataTransformFail", &deviceIntegerGenerator, "ResourceTestTransform_Fail", paramsTransformFail, true},
+		{"ErrorOccurredInDriver", &deviceIntegerGenerator, "Error", paramsError, true},
+	}
+	for _, tt := range tests {
+		t.Run(tt.testName, func(t *testing.T) {
+			if tt.testName == "MaxCmdOpsExceeded" {
+				common.CurrentConfig.Device.MaxCmdOps = 1
+				defer func() {
+					common.CurrentConfig.Device.MaxCmdOps = 128
+				}()
+			}
+			appErr := execWriteCmd(tt.device, tt.cmd, tt.params)
+			if !tt.expectErr && appErr != nil {
+				t.Errorf("%s expectErr:%v error:%v", tt.testName, tt.expectErr, appErr.Error())
+				return
+			}
+			if tt.expectErr && appErr == nil {
+				t.Errorf("%s expectErr:%v no error thrown", tt.testName, tt.expectErr)
+				return
+			}
+		})
+	}
+}
+
+func TestCommandAllHandler(t *testing.T) {
+	tests := []struct {
+		testName  string
+		cmd       string
+		body      string
+		method    string
+		expectErr bool
+	}{
+		{"PartOfReadCommandExecutionSuccess", "RandomValue_Uint8", "", methodGet, false},
+		{"PartOfReadCommandExecutionFail", "error", "", methodGet, true},
+		{"PartOfWriteCommandExecutionSuccess", "RandomValue_Uint8", `{"RandomValue_Uint8":"123"}`, methodSet, false},
+		{"PartOfWriteCommandExecutionFail", "error", `{"RandomValue_Uint8":"123"}`, methodSet, true},
+	}
+	for _, tt := range tests {
+		t.Run(tt.testName, func(t *testing.T) {
+			_, appErr := CommandAllHandler(tt.cmd, tt.body, tt.method)
+			if !tt.expectErr && appErr != nil {
+				t.Errorf("%s expectErr:%v error:%v", tt.testName, tt.expectErr, appErr.Error())
+				return
+			}
+			if tt.expectErr && appErr == nil {
+				t.Errorf("%s expectErr:%v no error thrown", tt.testName, tt.expectErr)
+				return
+			}
+		})
+	}
+}
+
+func TestCommandHandler(t *testing.T) {
+	var (
+		varsFindDeviceByValidId     = map[string]string{"id": "79ef304e-3b1e-4765-b509-798109c2467d", "command": "RandomValue_Int8"}
+		varsFindDeviceByInvalidId   = map[string]string{"id": "98621441-0f12-40ab-ad76-68edf1eeb4d5", "command": "RandomValue_Int8"}
+		varsFindDeviceByValidName   = map[string]string{"name": "Random-Integer-Generator01", "command": "RandomValue_Int8"}
+		varsFindDeviceByInvalidName = map[string]string{"name": "Random-Integer-Generator09", "command": "RandomValue_Int8"}
+		varsAdminStateLocked        = map[string]string{"name": "Random-Float-Generator01", "command": "RandomValue_Float32"}
+		varsProfileNotFound         = map[string]string{"name": "Random-Boolean-Generator01", "command": "error"}
+		varsCmdNotFound             = map[string]string{"name": "Random-Integer-Generator01", "command": "error"}
+		varsWriteInt8               = map[string]string{"name": "Random-Integer-Generator01", "command": "RandomValue_Int8"}
+	)
+	cache.Devices().UpdateAdminState("9a2b3ef2-6aed-41db-b868-79196c2a8b0a", contract.Locked)
```

Comment:
make this id as a public constant in mock_deviceclient.go and handle the returned error variable from cache